specs/ssh_configuration.adoc

```diff
@@ -0,0 +1,149 @@
+= SSH configuration for provisioning cluster
+
+SSH public-key configuration is used to authenticate cluster nodes with
+provisioning node.
+
+== Problem description
+
+The cluster provisioning process needs some communication between the provisioning
+node and cluster nodes. Before that the provisioning node needs some authentication
+to enable the secure communication with cluster nodes. SSH public-key configuration
+is the process of authenticating cluster nodes with provisioning node.
+
+== Use Cases
+
+Needs SSH configuration utility in common to support SSH configuration for
+provisioning cluster.
+
+== Proposed change
+
+SSH public-key configuration have four major process:
+
+1. Generate a key-pair for a user. The method can take the user name and
+   key-pair name as optional parameters, whose defaults are root and tendrl
+   respectively.
+2. Install an SSH key into a user's authorized_keys. SSH public key should be
+   a mandatory parameter, while the user name should be optional. User name
+   should default to root.
+3. Check if the SSH daemon is running and get it's port number.
+4. Check if the SSH incoming port is open.
```

Comment:
Not possible unless we support firewall configuration parsing. For now, we can only report on what port the sshd process listens on. Ensure that the port number and pid is returned from the method that checks for the running daemon.